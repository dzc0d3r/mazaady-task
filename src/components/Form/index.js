import Form from "./Form"
import Loading from "./Loading"

export { Form, Loading }
