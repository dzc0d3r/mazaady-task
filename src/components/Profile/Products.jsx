import { FiPlusCircle } from "react-icons/fi"
import { SingleProduct } from "@/components/Profile"

const Products = ({ products }) => {
  return (
    <div className="h-full w-full rounded-2xl bg-white p-5">
      <div className="flex flex-col items-start justify-between gap-2 md:flex-row md:items-center">
        <div className="flex flex-row gap-2">
          <button className="flex flex-row items-center gap-x-2 rounded-2xl border border-orange-500 bg-orange-200 bg-opacity-40 px-2 py-1 text-sm font-semibold text-orange-500 hover:to-orange-500 disabled:pointer-events-none disabled:opacity-50 md:px-5 md:py-2 md:text-base">
            Products
          </button>
          <button className="flex flex-row items-center gap-x-2 rounded-2xl border border-gray-300 bg-white/30 px-2 py-1 text-sm font-semibold text-gray-500 disabled:pointer-events-none disabled:opacity-50 md:px-5  md:py-2 md:text-base">
            Articles
          </button>
          <button className="flex flex-row items-center gap-x-2 rounded-2xl border border-gray-300 bg-white/30 px-2 py-1 text-sm font-semibold text-gray-500 disabled:pointer-events-none disabled:opacity-50 md:px-5  md:py-2 md:text-base">
            Reviews
          </button>
        </div>
        <button className="flex flex-row items-center gap-x-2 rounded-xl bg-gradient-to-r from-primary to-orange-600 px-2 py-2 text-sm font-semibold text-white hover:to-orange-500  disabled:pointer-events-none disabled:opacity-50 md:px-5 md:py-2 md:text-base">
          <FiPlusCircle className="text-lg md:text-2xl" />
          Add Review
        </button>
      </div>

      <div className="relative my-5 flex flex-col gap-2">
        <h1 className=" text-3xl font-semibold text-black/80 md:text-4xl ">
          Products
          <span className="absolute top-0 mx-3 text-base font-light  text-gray-700">
            (12)
          </span>
        </h1>
        <SingleProduct products={products} />
      </div>
    </div>
  )
}

export default Products
