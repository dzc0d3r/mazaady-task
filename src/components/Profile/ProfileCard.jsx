import Image from "next/image"
import { TbUserCheck, TbUsers } from "react-icons/tb"
import { RiMagicLine } from "react-icons/ri"
import { profileData } from "@/lib/data"

const ProfileCard = ({ fullname, bio, followers, following, rate }) => {
  return (
    <div className="mx-auto h-fit w-full max-w-full rounded-2xl bg-white p-5 md:max-w-none">
      <div className="flex flex-col">
        <Image
          src={profileData.imageSrc}
          height={100}
          width={100}
          alt="Profile image"
          className="h-24 w-24 rounded-2xl"
        ></Image>
      </div>
      <div className="mt-5">
        <h2 className="text-2xl font-semibold text-black/70">{fullname}</h2>
        <p className="mt-2 text-sm font-light text-black/70">{bio}</p>
      </div>
      <div className="my-5 flex flex-row gap-2">
        {/* stats */}
        <div className="flex w-full gap-1 rounded-2xl bg-orange-400 bg-opacity-10 p-2">
          {/* following */}
          <div className="flex items-center justify-center">
            <TbUserCheck className="text-2xl text-orange-400 md:text-3xl" />
          </div>
          <div className="flex flex-col">
            <p className="text-sm font-bold text-black/70 md:text-base">
              {following}
            </p>
            <span className="text-xs capitalize text-orange-400  md:text-sm">
              following
            </span>
          </div>
        </div>
        <div className="flex w-full gap-1 rounded-2xl bg-orange-400 bg-opacity-10 p-2">
          {/* followers */}
          <div className="flex items-center justify-center">
            <TbUsers className="text-2xl text-orange-400 md:text-3xl" />
          </div>
          <div className="flex flex-col">
            <p className="text-sm font-bold text-black/70 md:text-base">
              {followers}
            </p>
            <span className="text-xs capitalize text-orange-400  md:text-sm">
              followers
            </span>
          </div>
        </div>
        <div className="flex w-full gap-1 rounded-2xl bg-orange-400 bg-opacity-10 p-2">
          {/* following */}
          <div className="flex items-center justify-center">
            <RiMagicLine className="text-2xl font-semibold text-orange-400 md:text-3xl " />
          </div>
          <div className="flex flex-col">
            <p className="text-sm font-bold text-black/70 md:text-base">
              {rate.value}
              <span className="mx-2 text-xs font-light text-gray-600 md:text-sm">
                {rate.count}
              </span>
            </p>
            <span className="text-xs capitalize text-orange-400  md:text-sm">
              rate
            </span>
          </div>
        </div>
      </div>
      <button className="mt-5 flex w-full flex-row items-center justify-center gap-x-2 rounded-xl bg-gradient-to-r from-primary to-orange-600 px-4 py-2 text-center text-lg   text-white hover:to-orange-500 disabled:pointer-events-none disabled:opacity-50">
        Follow
      </button>
    </div>
  )
}

export default ProfileCard
