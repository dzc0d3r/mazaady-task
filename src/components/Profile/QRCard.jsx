"use client"
import { IoEyeOutline } from "react-icons/io5"
import { GoShareAndroid } from "react-icons/go"
import { PiFileArrowDown } from "react-icons/pi"
import { TfiArrowCircleDown } from "react-icons/tfi"
import { useState } from "react"
import Image from "next/image"

const QRCard = () => {
  const [collapse, setCollapse] = useState(false)
  return (
    <div className="mx-auto flex h-fit w-full max-w-full flex-col rounded-2xl bg-white p-5 md:max-w-none">
      <div className="flex flex-row items-center justify-between">
        <h2 className="text-2xl font-semibold capitalize text-black/70">
          QR Code
        </h2>
        <div className="group flex flex-row gap-2">
          <IoEyeOutline className="text-2xl text-black/70 group-hover:cursor-pointer" />
          <GoShareAndroid className="text-2xl text-black/70 group-hover:cursor-pointer" />
          <PiFileArrowDown className="text-2xl text-black/70 group-hover:cursor-pointer" />
          <div className=" relative inline-flex lg:hidden">
            <input
              onClick={() => setCollapse(!collapse)}
              type="checkbox"
              id="faq1"
              className="peer absolute left-0 top-0 z-10 h-full w-full appearance-none text-center hover:cursor-pointer"
            />
            <TfiArrowCircleDown className="rounded-full bg-orange-200 bg-opacity-60 text-2xl text-black/70 " />
          </div>
        </div>
      </div>
      <div className={`${collapse ? "hidden" : "mt-4 flex flex-col"}`}>
        <div className="flex flex-row items-center gap-3 rounded-xl bg-orange-200 bg-opacity-40 p-5">
          <PiFileArrowDown className="text-3xl text-orange-400" />
          <p className="text-xs font-light text-black/70">
            Download the QR code or share it with your friends.
          </p>
        </div>
        <div className="mt-5 flex h-full w-full justify-center">
          <Image
            src={"qr.svg"}
            width={100}
            height={100}
            alt="Profile image"
            className="h-80 w-96 rounded-2xl"
          ></Image>
        </div>
      </div>
    </div>
  )
}

export default QRCard
