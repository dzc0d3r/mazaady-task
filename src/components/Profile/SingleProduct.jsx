import Image from "next/image"
import { PiHeart } from "react-icons/pi"
import { PiHeartFill } from "react-icons/pi"

const SingleProduct = ({ products }) => {
  return (
    <>
      {products.map((product, index) => (
        <div key={index} className="my-2 grid grid-cols-4 gap-4 p-0 md:p-4">
          <div className="group relative col-span-1 h-24 w-24 hover:cursor-pointer md:h-40 md:w-40">
            <Image
              src={product.imageSrc}
              fill
              alt="Profile image"
              className="absolute cursor-pointer rounded-2xl transition-all duration-300 group-hover:scale-[1.01] "
            ></Image>
            <div
              className={`-botton-px absolute right-0 flex md:-right-px md:bottom-[.55rem]  
                            ${product.tag.includes("live") ? "bg-primary" : "bg-orange-500"} h-fit w-fit rounded-ee-full rounded-ss-full px-2 py-[2px] text-[.6rem]  font-light capitalize text-white md:px-6 md:py-2 lg:text-sm`}
            >
              <p>{product.tag}</p>
            </div>
          </div>
          <div className="col-span-3 mx-7 mt-1 flex flex-row items-start md:mx-1 md:mt-4 md:justify-between">
            <div className="flex flex-col gap-3 ">
              <h3 className="max-w-52 break-words break-all text-sm font-medium text-black/70 lg:max-w-full lg:text-lg">
                {product.title}
              </h3>

              <p className="text-xs font-light text-black/70 md:mt-1 md:text-base ">
                Starting Price{" "}
                <span className="text-lg font-semibold text-black/80 md:text-2xl">
                  {product.starting_price} EGP
                </span>
              </p>
              <div className="flex flex-col items-start gap-2 md:flex-row md:items-center">
                <p className="flex items-center gap-2 text-xs font-light text-black/70 md:mt-1 md:flex-row md:text-base ">
                  Lot Starts In
                </p>
                <div className="flex flex-row gap-1 md:gap-2">
                  <span className="rounded-2xl bg-orange-200 px-2 py-[6px] text-xs font-semibold text-orange-500  md:px-4 md:py-1 md:text-xl">
                    {product.start_in.days}
                    <span className="mx-1 text-[.6rem] font-medium md:text-sm">
                      Days
                    </span>
                  </span>
                  <span className="rounded-2xl bg-orange-200 px-2 py-[6px] text-xs font-semibold text-orange-500  md:px-4 md:py-1 md:text-xl">
                    {product.start_in.hours}
                    <span className="mx-1 text-[.6rem] font-medium md:text-sm">
                      Hours
                    </span>
                  </span>
                  <span className="rounded-2xl bg-orange-200 px-2 py-[6px] text-xs font-semibold text-orange-500  md:px-4 md:py-1 md:text-xl">
                    {product.start_in.minutes}
                    <span className="mx-1 text-[.6rem] font-medium md:text-sm">
                      Minutes
                    </span>
                  </span>
                </div>
              </div>
            </div>
            {/* heart icon */}
            {product.liked ? (
              <PiHeartFill className="text-3xl text-primary transition-all duration-300 hover:scale-105 hover:cursor-pointer" />
            ) : (
              <PiHeart className="text-3xl text-gray-500 transition-all duration-300 hover:scale-105 hover:cursor-pointer" />
            )}
          </div>
        </div>
      ))}
    </>
  )
}

export default SingleProduct
