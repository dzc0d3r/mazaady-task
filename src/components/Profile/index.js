import ProfileCard from "./ProfileCard"
import QRCard from "./QRCard"
import Products from "./Products"
import SingleProduct from "./SingleProduct"

export { ProfileCard, QRCard, Products, SingleProduct }
