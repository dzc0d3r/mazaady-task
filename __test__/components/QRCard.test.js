import React from "react"
import renderer from "react-test-renderer"
import { render, screen, fireEvent } from "@testing-library/react"
import { QRCard } from "@/components/Profile"

describe("QRCard Component", () => {
  it("renders QRCard component correctly", () => {
    // snapshot test
    const tree = renderer.create(<QRCard />).toJSON()
    expect(tree).toMatchSnapshot()

    // Check if the component is rendered
    render(<QRCard />)
    const qrCardTitle = screen.getByText("QR Code")
    expect(qrCardTitle).toBeInTheDocument()

    // Check if the checkbox is rendered
    const checkbox = screen.getByRole("checkbox")
    expect(checkbox).toBeInTheDocument()

    // Check if the default collapse state is not collapsed
    const qrCodeContent = screen.getByText(
      "Download the QR code or share it with your friends.",
    )
    expect(qrCodeContent).toBeInTheDocument()
  })

  it("toggles collapse state when checkbox is clicked", () => {
    render(<QRCard />)

    // Check if the component is initially not collapsed
    let qrCodeContent = screen.getByText(
      "Download the QR code or share it with your friends.",
    )
    expect(qrCodeContent).toBeInTheDocument()

    // Click the checkbox to collapse the component
    const checkbox = screen.getByRole("checkbox")
    fireEvent.click(checkbox)

    // Check if the component is collapsed after clicking the checkbox
    qrCodeContent = screen.queryByText(
      "Download the QR code or share it with your friends.",
    )
    const parentOfParent = qrCodeContent.parentElement?.parentElement
    expect(parentOfParent).toHaveClass("hidden")
  })

  it("renders the QR image when not collapsed", () => {
    render(<QRCard />)

    // Check if the QR image is initially rendered
    const qrImage = screen.getByAltText("Profile image")
    expect(qrImage).toBeInTheDocument()
  })

  it("does not render the QR image when collapsed", () => {
    render(<QRCard />)

    // Click the checkbox to collapse the component
    const checkbox = screen.getByRole("checkbox")
    fireEvent.click(checkbox)

    // Check if the QR image is not rendered after collapsing
    const qrImage = screen.queryByAltText("Profile image")
    const parentOfParent = qrImage.parentElement?.parentElement
    expect(parentOfParent).toHaveClass("hidden")

    // Add more tests as needed based on your component's behavior
  })
})
