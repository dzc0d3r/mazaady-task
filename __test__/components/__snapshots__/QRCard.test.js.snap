// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCard Component renders QRCard component correctly 1`] = `
<div
  className="mx-auto flex h-fit w-full max-w-full flex-col rounded-2xl bg-white p-5 md:max-w-none"
>
  <div
    className="flex flex-row items-center justify-between"
  >
    <h2
      className="text-2xl font-semibold capitalize text-black/70"
    >
      QR Code
    </h2>
    <div
      className="group flex flex-row gap-2"
    >
      <svg
        className="text-2xl text-black/70 group-hover:cursor-pointer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M255.66 112c-77.94 0-157.89 45.11-220.83 135.33a16 16 0 0 0-.27 17.77C82.92 340.8 161.8 400 255.66 400c92.84 0 173.34-59.38 221.79-135.25a16.14 16.14 0 0 0 0-17.47C428.89 172.28 347.8 112 255.66 112z"
          fill="none"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="32"
        />
        <circle
          cx="256"
          cy="256"
          fill="none"
          r="80"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
      </svg>
      <svg
        className="text-2xl text-black/70 group-hover:cursor-pointer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 5.5a3.498 3.498 0 0 1-6.062 2.385l-5.112 3.021a3.498 3.498 0 0 1 0 2.188l5.112 3.021a3.5 3.5 0 1 1-.764 1.29l-5.112-3.02a3.499 3.499 0 1 1-3.843-5.642 3.499 3.499 0 0 1 3.843.872l5.112-3.021A3.5 3.5 0 1 1 20 5.5Zm-1.5 13a2 2 0 1 0-3.999-.001 2 2 0 0 0 3.999.001Zm0-13a2 2 0 1 0-3.999-.001A2 2 0 0 0 18.5 5.5ZM5.5 14a2 2 0 1 0 .001-3.999A2 2 0 0 0 5.5 14Z"
        />
      </svg>
      <svg
        className="text-2xl text-black/70 group-hover:cursor-pointer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,82.34l-56-56A8,8,0,0,0,152,24H56A16,16,0,0,0,40,40V216a16,16,0,0,0,16,16H200a16,16,0,0,0,16-16V88A8,8,0,0,0,213.66,82.34ZM160,51.31,188.69,80H160ZM200,216H56V40h88V88a8,8,0,0,0,8,8h48V216Zm-42.34-61.66a8,8,0,0,1,0,11.32l-24,24a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L120,164.69V120a8,8,0,0,1,16,0v44.69l10.34-10.35A8,8,0,0,1,157.66,154.34Z"
        />
      </svg>
      <div
        className=" relative inline-flex lg:hidden"
      >
        <input
          className="peer absolute left-0 top-0 z-10 h-full w-full appearance-none text-center hover:cursor-pointer"
          id="faq1"
          onClick={[Function]}
          type="checkbox"
        />
        <svg
          className="rounded-full bg-orange-200 bg-opacity-60 text-2xl text-black/70 "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          version="1.1"
          viewBox="0 0 17 17"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g />
          <path
            d="M12.146 6.146l0.707 0.707-4.353 4.354-4.354-4.353 0.707-0.707 3.647 3.646 3.646-3.647zM17 8.5c0 4.687-3.813 8.5-8.5 8.5s-8.5-3.813-8.5-8.5 3.813-8.5 8.5-8.5 8.5 3.813 8.5 8.5zM16 8.5c0-4.136-3.364-7.5-7.5-7.5s-7.5 3.364-7.5 7.5 3.364 7.5 7.5 7.5 7.5-3.364 7.5-7.5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="mt-4 flex flex-col"
  >
    <div
      className="flex flex-row items-center gap-3 rounded-xl bg-orange-200 bg-opacity-40 p-5"
    >
      <svg
        className="text-3xl text-orange-400"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M213.66,82.34l-56-56A8,8,0,0,0,152,24H56A16,16,0,0,0,40,40V216a16,16,0,0,0,16,16H200a16,16,0,0,0,16-16V88A8,8,0,0,0,213.66,82.34ZM160,51.31,188.69,80H160ZM200,216H56V40h88V88a8,8,0,0,0,8,8h48V216Zm-42.34-61.66a8,8,0,0,1,0,11.32l-24,24a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L120,164.69V120a8,8,0,0,1,16,0v44.69l10.34-10.35A8,8,0,0,1,157.66,154.34Z"
        />
      </svg>
      <p
        className="text-xs font-light text-black/70"
      >
        Download the QR code or share it with your friends.
      </p>
    </div>
    <div
      className="mt-5 flex h-full w-full justify-center"
    >
      <img
        alt="Profile image"
        className="h-80 w-96 rounded-2xl"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="qr.svg"
        style={
          {
            "color": "transparent",
          }
        }
        width={100}
      />
    </div>
  </div>
</div>
`;
